release/v1alpha1/types.go

```diff
@@ -0,0 +1,76 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PromotionTestResults encapsulates the inputs to produce a new set of promotion tests and
```

Comment:
no it doesn't.  Reading this, it looks like a way to aggregate the outputs.

Reply:
Tell me about names, any fixed labels, how I map from imagestreams (imagestreamtags?) to this and back again.

Reply:
explicitly state your intent about namespaced or not namespaced.

Reply:
My initial instinct says we should probably be namespaced.  I added the `// +genclient` and it looks like you should only add the `// +genclient:nonNamespaced` if you want cluster scoped resources.  Is there anything more that's necessary?

Reply:
As for the names, labels, and mapping to/from imagestreams, that all seems like it's higher level than the `PromotionTestResults`.  Potentially something more like what @stevekuznetsov was thinking with the `ReleasePayload`?